pkg/cmd/list/edge_applications/edge_applications.go

```diff
@@ -0,0 +1,116 @@
+package edge_applications
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+
+	"github.com/MakeNowJust/heredoc"
+	table "github.com/MaxwelMazur/tablecli"
+	"github.com/aziontech/azion-cli/messages/general"
+	msg "github.com/aziontech/azion-cli/messages/list/edge_applications"
+	api "github.com/aziontech/azion-cli/pkg/api/edge_applications"
+	"github.com/aziontech/azion-cli/pkg/cmdutil"
+	"github.com/aziontech/azion-cli/pkg/contracts"
+	"github.com/aziontech/azion-cli/utils"
+	"github.com/spf13/cobra"
+)
+
+func NewCmd(f *cmdutil.Factory) *cobra.Command {
+	opts := &contracts.ListOptions{}
+
+	cmd := &cobra.Command{
+		Use:           msg.ListUsage,
+		Short:         msg.ListShortDescription,
+		Long:          msg.ListLongDescription,
+		SilenceUsage:  true,
+		SilenceErrors: true, Example: heredoc.Doc(`
+            $ azion list edge_applications
+            $ azion list edge_applications --details
+            $ azion list edge_applications --page 1 
+            $ azion list edge_applications --page_size 5
+        `),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			var numberPage int64 = opts.Page
+
+			client := api.NewClient(f.HttpClient,
+				f.Config.GetString("api_url"),
+				f.Config.GetString("token"),
+			)
+
+			if !cmd.Flags().Changed("page") && !cmd.Flags().Changed("page_size") {
+				for {
+					pages, err := PrintTable(client, f, opts, &numberPage)
+					if numberPage > pages && err == nil {
+						return nil
+					}
+					if err != nil {
+						return fmt.Errorf(msg.ErrorGetAll.Error(), err)
+					}
+				}
+			}
+
+			if _, err := PrintTable(client, f, opts, &numberPage); err != nil {
```

Comment:
Am I crazy or PrintTable is being run twice?

Reply:
you're not crazy it's exactly that and two flows, but it's a functional and valid logic, this is a good time to refactor hehe  🖍️